evm-adapters/src/sputnik/cheatcodes/memory_stackstate_owned.rs

```diff
@@ -47,6 +47,8 @@ pub struct MemoryStackStateOwned<'config, B> {
     pub accesses: Option<RecordAccess>,
     pub all_logs: Vec<String>,
     pub expected_emits: Vec<ExpectedEmit>,
+    pub debug_enabled: bool,
+    pub debug_steps: Vec<DebugArena>,
```

Comment:
Similarly, i think it might be hard, but refactoring to `TracingMemStackStateOwned`, `CheatMemStackStateOwned` and `DebugMemStackStateOwned` would let the code be a lot cleaner, vs having 1 gigantic struct which stores everything for all execution modes and `_enabled` bools

Reply:
I agree. Youre convincing me but on the other side of the argument is I am really not a fan of 10 levels of indirection. No one can ever keep it straight in their head and smells like C or C++ code. I'll give some thought on compromises/cleanup